components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/extension/eidas/EidasExtensionProcessor.java

```diff
@@ -0,0 +1,354 @@
+import org.opensaml.saml2.core.Assertion;
+import org.opensaml.saml2.core.AttributeStatement;
+import org.opensaml.saml2.core.AuthnContextComparisonTypeEnumeration;
+import org.opensaml.saml2.core.AuthnRequest;
+import org.opensaml.saml2.core.NameID;
+import org.opensaml.saml2.core.RequestAbstractType;
+import org.opensaml.saml2.core.Response;
+import org.opensaml.saml2.core.StatusResponseType;
+import org.opensaml.saml2.core.impl.NameIDBuilder;
+import org.opensaml.xml.XMLObject;
+import org.opensaml.xml.schema.impl.XSAnyImpl;
+import org.w3c.dom.NodeList;
+import org.wso2.carbon.identity.application.common.model.Claim;
+import org.wso2.carbon.identity.application.common.model.ClaimMapping;
+import org.wso2.carbon.identity.base.IdentityException;
+import org.wso2.carbon.identity.sso.saml.SAMLSSOConstants;
+import org.wso2.carbon.identity.sso.saml.dto.SAMLSSOAuthnReqDTO;
+import org.wso2.carbon.identity.sso.saml.dto.SAMLSSOReqValidationResponseDTO;
+import org.wso2.carbon.identity.sso.saml.exception.IdentitySAML2SSOException;
+import org.wso2.carbon.identity.sso.saml.extension.SAMLExtensionProcessor;
+import org.wso2.carbon.identity.sso.saml.extension.eidas.model.RequestedAttributes;
+import org.wso2.carbon.identity.sso.saml.extension.eidas.model.SPType;
+import org.wso2.carbon.identity.sso.saml.extension.eidas.util.EidasConstants;
+import org.wso2.carbon.identity.sso.saml.util.SAMLSSOUtil;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.apache.commons.collections.CollectionUtils.isNotEmpty;
+
+/**
+ * This class is used to process and validate the eIDAS SAML extensions.
+ */
+public class EidasExtensionProcessor implements SAMLExtensionProcessor {
+    private static Log log = LogFactory.getLog(EidasExtensionProcessor.class);
+    private static String errorMsg = "Mandatory Attribute not found.";
+
+    /**
+     * Check whether the SAML authentication request can be handled by the eIDAS extension processor.
+     *
+     * @param request SAML request
+     * @return true if the request can be handled
+     * @throws IdentitySAML2SSOException
+     */
+    @Override
+    public boolean canHandle(RequestAbstractType request) throws IdentitySAML2SSOException {
+
+        return request.getNamespaces().stream().anyMatch(namespace -> namespace.getNamespaceURI()
+                .equals(EidasConstants.EIDAS_NS));
+    }
+
+    /**
+     * Check whether the SAML response can be handled by the eIDAS extension processor.
+     *
+     * @param authReqDTO Authentication request data object
+     * @return true if the request can be handled
+     * @throws IdentitySAML2SSOException
+     */
+    @Override
+    public boolean canHandle(StatusResponseType response, Assertion assertion, SAMLSSOAuthnReqDTO authReqDTO)
+            throws IdentitySAML2SSOException {
+
+        return authReqDTO.getProperties().getProperty(EidasConstants.EIDAS_REQUEST)
```

Comment:
The DTO should have the capability get a single property or all properties.
Wrap the property list within the DTO

Reply:
Better to add a debug log on whether the respective extension can handle the response